src/internal/storage/fileset/storage.go

```diff
@@ -328,30 +274,60 @@ func (s *Storage) GC(ctx context.Context) error {
 	return gc.Run(ctx)
 }
 
-func (s *Storage) levelSize(i int) int64 {
-	return s.levelZeroSize * int64(math.Pow(float64(s.levelSizeBase), float64(i)))
-}
-
-const subFileSetFmt = "%020d"
-const levelFmt = "level_" + subFileSetFmt
-
-// SubFileSetStr returns the string representation of a subfileset.
-func SubFileSetStr(subFileSet int64) string {
-	return fmt.Sprintf(subFileSetFmt, subFileSet)
+func (s *Storage) newPrimitive(ctx context.Context, prim *Primitive, ttl time.Duration) (*ID, error) {
+	id := newID()
+	md := &Metadata{
+		Value: &Metadata_Primitive{
+			Primitive: prim,
+		},
+	}
+	var pointsTo []string
+	for _, chunkID := range prim.PointsTo() {
+		pointsTo = append(pointsTo, chunk.ObjectID(chunkID))
+	}
+	if err := s.tracker.CreateObject(ctx, filesetObjectID(id), pointsTo, ttl); err != nil {
```

Comment:
It would be good if we could get the object creation and store set in the same transaction.

Reply:
I think we can get by without transactions, I have a ticket for a helper function that creates a temporary object, add the auxilary data, then upgrade the object to permanent.   That workflow prevents leaks, and doesn't need transactions.

Reply:
Okay nevermind. We do need transactions, or at least it is easier with them.  There is a TODO in the driver, and after the tracker supports `sql.Tx` all these spots will be replaced.